mlflow/system_metrics/metrics/gpu_monitor.py

```diff
@@ -0,0 +1,48 @@
+"""Class for monitoring GPU stats."""
+
+import logging
+import sys
+
+from mlflow.system_metrics.metrics.base_metrics_monitor import BaseMetricsMonitor
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import pynvml
+except ImportError:
+    # If `pynvml` is not installed, a warning will be logged at monitor instantiation.
```

Comment:
Should we install `pynvml` in MLR? It's not installed currently:

https://docs.databricks.com/en/release-notes/runtime/13.3lts-ml.html